scripts/link-packages.js

```diff
@@ -0,0 +1,176 @@
+  }
+
+  const spawn = spawnSync('yarn', ['link'], {
+    cwd: pkg.dir,
+  });
+
+  if (spawn.stdout.exit === 1) {
+    console.log(`${pkg.name} failed to link!`);
+  }
+
+  if (!silent) {
+    console.log(`${pkg.name} - linked`);
+  }
+}
+
+/**
+ * Links package to another package
+ *
+ * @param { name, file, dir } linkPkg Package that will be linked
+ * @param { name, file, dir } targetPkg Target package
+ * @return {undefined}
+ */
+function packageConnector(linkPkg, targetPkg) {
+  if (!linkPkg && !targetPkg) {
+    throw new Error('No packages provided to connect!');
+  }
+
+  const spawn = spawnSync('yarn', ['link', linkPkg.name], {
+    cwd: targetPkg.dir,
+  });
+
+  const spawnResponse = spawn.stdout.toString('utf8');
+
+  const message = `${linkPkg.name} => ${targetPkg.name}`;
+
+  if (spawnResponse.indexOf('yarnpkg.com') >= 0) {
+    console.error(`Error: ${message} `);
+  } else {
+    console.log(message);
+  }
+}
+
+/**
+ * Extract recusively all `package.json`s inside directory
+ *
+ * @param {string} directory absolute directory location
+ * @param {Array<{ name, file, dir }>} files list of package.json
+ * @return {Array<{ name, file, dir }>} list of package.json
+ */
+function extractRootPackageJsons(directory, files) {
+  // Excludes others than the built packages
+  if (directory.indexOf('node_modules') >= 0) {
+    return;
+  }
+
+  readdirSync(directory).forEach(file => {
+    const absolute = join(directory, file);
+
+    if (statSync(absolute).isDirectory()) {
+      return extractRootPackageJsons(absolute, files);
+    }
+
+    // Returns only folders that contains `package.json`
+    if (file === 'package.json') {
+      const json = require(absolute);
+
+      const isPkgRoot = isRootPackage(json);
+
+      if (!isPkgRoot) {
+        return files;
+      }
+
+      // Add root package.json to the list
+      return files.push({
+        name: json.name,
+        file: absolute,
+        dir: directory,
+      });
+    }
+  });
+
+  return files;
+}
+
+const packageJsons = extractRootPackageJsons(process.cwd(), []); // Collect every root package.json inside the monorepo
+
+if (!packageJsons) {
+  console.log('Warning: No packages available!');
+  return;
+}
+
+const distPackages = packageJsons.filter(x => x.dir.indexOf(distFolder) >= 0);
+
+const packages = packageJsons.filter(x => x.dir.indexOf('dist') < 0);
+
+// Link all packages in dist
+distPackages.forEach(pkg => linkPackage(pkg));
+
+packages.forEach(pkg => {
+  const packageJson = require(pkg.file);
+
+  // Escapes package that doesn't have dependecies
+  if (!packageJson.dependencies) {
+    return;
+  }
+
+  /**
+   * Checks if the currenct package depends on other packages in the mono repo and links them
+   *
+   * e.g. Website -> @clr/core
+   * e.g. @clr/angular -> @clr/core
+   */
+  distPackages.forEach(distPkg => {
+    if (!packageJson.dependencies[distPkg.name]) {
+      return;
+    }
+
+    packageConnector(distPkg, pkg);
+  });
+});
```

Comment:
Do we ever have to worry about unlinking or are the link removed when the dist is deleted?

Reply:
the link is removed when the `dist` is deleted, yarn even notifies that when the build script runs with something like `link for package {smth} was removed`